src/Integration/Binding/BindingWorkflow.cs

```diff
@@ -125,8 +126,8 @@ private ProgressStepDefinition[] CreateWorkflowSteps(IProgressController control
                 // * create/update per-project ruleset
                 // * set project-level properties
                 // Most of the work is delegated to SolutionBindingOperation
-                new ProgressStepDefinition(Strings.BindingProjectsDisplayMessage, IndeterminateNonCancellableUIStep,
-                        (token, notifications) => this.InitializeSolutionBindingOnUIThread(notifications)),
+                new ProgressStepDefinition(Strings.BindingProjectsDisplayMessage, IndeterminateNonCancellableBackgroundStep,
```

Comment:
The only necessary change to fix the binding code is to change this step to start on the background thread rather than the UI thread.

The `UnboundProjectFinder` then handles switching on and off the UI thread when necessary.